### ISSUE TITLE
accuracy function for analogy test gives back empty lists as result

### ISSUE THREAD

**Issue by vlfgns5** [RaRe-Technologies/gensim#2455]
#### Problem description
Analogy test for accuracy function returns empty list due to typo. (topn = False)

#### Solution
There are 2 solutions regarding this matter in my opinion
: First one is the change  keyedvectors.py script in gensim/models on line 1187

Line : 1187 in keyedvectors.py
(https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/keyedvectors.py)

sims = most_similar(self, positive=[b, c], negative=[a], topn=False, restrict_vocab=restrict_vocab)
==>
sims = most_similar(self, positive=[b, c], negative=[a], topn=None, restrict_vocab=restrict_vocab)

Second solution
: change keyedvectors.py script in gensim/models on line 514, 556

if topn is not None and topn < 1:
==>
if not topn and topn<1:

&&

if topn is None:
==>
if not topn:

Well...first one is better i think

**Comment by piskvorky** [comment]
Thanks for reporting @vlfgns5.

The problem seems to have been introduced by #2356. @Witiko can you have a look? Are there other places similarly affected by that change?

**Comment by Witiko** [comment]
@piskvorky You are correct. Where previously topn was only cast to a boolean, #2356 gave `topn=0` different semantics (return an empty iterable) than `topn=None` (return all similarities, previously undocumented behavior). However, I missed callers that would use `topn=False`.

According to `grep topn=False`, there are no callers other than `EuclideanKeyedVectors.accuracy` that use `topn=False`. I can either fix the called by changing `topn=False` to `topn=None`, or I can make `most_similar` accept both `False` and `None`.

**Comment by Witiko** [comment]
Digging deeper into this, properly supporting `most_similar(…, topn=None)` needs more work. When `indexer` is `None`, then `topn=None` (and `topn=False`) returns the expected result. However, when an `indexer` is provided (such as `Annoyindexer`), then the call fails, because Annoy only supports integer `topn`. Therefore, we need an additional fix that disables the indexer when `topn` is None:

```diff
< if indexer is not None:
> if indexer is not None and isinstance(topn, int):
```
